### PR TITLE
POC Dynamic types

### PR DESCRIPTION
**Pull request by kaare**
Comments, please. Proof Of Concept - This is the way I'd like to go with types, but a lot of fiddling still remains. E.g. 

- More p6ifying.
  Can I pass a type object to a setter and use as hash key?
  Can I avoid the need for explicit passing the type to the sub?
  How to use a Str to create a type? (May not be necessary if the above is solved

(Perhaps in a separate module) I may write something that find the oid of an enum, or a new type

The two new test files show how to change type conversions.

### CODE REVIEW

File: lib/DBDish.pm6 [hunk 1]
```diff
@@ -19,6 +19,17 @@ role Driver does DBDish::ErrorHandling {
     #method new() { nextsame; }
 }
 
+role Type {
+    has Sub %!Conversions{Str};
```
Comment (jonathanstowe) [inline lib/DBDish.pm6]: This probably wants to be relaxed to Routine or even Callable, using Sub here is a bit specific I think :)

File: lib/DBDish.pm6 [hunk 2]
```diff
@@ -19,6 +19,17 @@ role Driver does DBDish::ErrorHandling {
...
+
+    method set(Str $name, Sub $convert) {
```
Comment (jonathanstowe) [inline lib/DBDish.pm6]: I'd just adjust the ```$convert``` to ```&convert``` and lose the explicit type constraint personally (that way the caller is free to implement it how they want as long as it supports Callable semantics.)

**Comment by jonathanstowe** [comment]
I'm not sure what the criteria for merging this are going to be now.

Maybe we should bring it into its own branch here, like we did with the Oracle work before so we can keep it up to date and so forth.

**Comment by jonathanstowe** [comment]
Or just merge it like that. Nice one.

**Comment by zoffixznet** [comment]
Based on the error message I get, looks like this merge broke installation of the module. See https://github.com/perl6/DBIish/issues/85

**Pull request merged**